### ISSUE TITLE
Firefox: hanging on 75%, HD doesn't buffer right

### ISSUE THREAD

**Issue by MichelvanderAa** [robbert/book-of-sand#122]
Over the last few days I've received quite a few comments that the site doesn't work well on Firefox. Some people had the old 'hanging on 75%' bug during start-up. 
Some others complained that when switching to HD the video just stalls and audio keeps looping. I experience the same bug here at home. If you select HD and jump forward in time the buffering code doesn't seem to be able to lock in.

So I guess there's two options;  A)try fix the bug  B)add Firefox to the list of unsupported browsers and send them to a screen that suggests to try again with Chrome.

**Comment by robbert** [comment]
I'll look into the issues as soon as I've got time, which probably is on Saturday.

**Comment by MichelvanderAa** [comment]
ok thx

**Comment by MichelvanderAa** [comment]
At a family gathering today two more people said they couldn't load the site, both turned out to be using Firefox. So until this bug is fixed I feel we should redirect Firefox users to a page that says "your browser is not supported, please use Google Chrome". (or something similar)

**Comment by robbert** [comment]
We discussed this before, but would it be possible for you to encode the video in WebM as well?
After some investigation, I think this might very well fix this issue. I think some installs of Firefox don't support the video because of software patents in the US. Supposedly, encoding in WebM VP8 would support Firefox 4 and up, VP9 would be Firefox 28 and up.

**Comment by MichelvanderAa** [comment]
happy to try but the FFmpeg app I appear to need to encode is residing in nerdville.... no idea how and where to download... this link below just gives me a folder with weird stuff..
https://github.com/FFmpeg/FFmpeg

**Comment by robbert** [comment]
I'm not sure what software you're using, but is this perhaps what you're looking for?
http://www.fnordware.com/WebM/

**Comment by MichelvanderAa** [comment]
I use Final Cut Pro 7.  Also happy to try a stand-alone encoder or Quicktime library but can't find a proper install of FFmpeg

**Comment by MichelvanderAa** [comment]
OK I managed to encode the video to WebM/VP8. I've made a SD file and HD file. They're on our S3:
- Firefox HD 1280x1950_15juni_WebM.mov
- Firefox SD 720x1098_15juni_WebM.mov

You still need to set the metadata right, I don't know what the right video metadata for this codec is

Can you create a code on the beta website so that Firefox users get these two video files? If it's in place let me know and I'll send some beta testers to try! thx

**Comment by robbert** [comment]
The encoded video doesn't work for me in Firefox, while it does work in other browsers. Does it play in your Firefox? You need to rename it to `something.webm` instead of `.mov` for this to work, then drag-and-drop it in a Firefox window.

![screen shot 2015-06-16 at 21 14 37](https://cloud.githubusercontent.com/assets/30694/8192237/1e3bd91e-146d-11e5-9593-83c579d2bbff.png)

Other `.webm` videos play fine in Firefox.

**Comment by robbert** [comment]
Alternatively, we can also try encoding in `.ogg` if Final Cut Pro is able to do that.

**Comment by robbert** [comment]
Or maybe it's calld `.ogv`

**Comment by MichelvanderAa** [comment]
it's definitely VP8 codec:
![screen shot 2015-06-16 at 21 38 35](https://cloud.githubusercontent.com/assets/12616308/8192681/2b193aac-1470-11e5-8dc4-c39e9d122ad6.png)

took me 30 hrs to encode :-(

**Comment by robbert** [comment]
Can you make some screenshots of the encoding settings available for WebM? Does it say something about Matroska or MKV containers?

**Comment by robbert** [comment]
When I look at the bits and bytes of your files and some other WebM files, the other files seem to start with `1a45 dfa3 a342` and your files don't. Maybe the video is correctly encoded as WebM stream, but the container format isn't correct somehow. Maybe it is possible to encode WebM in a `.mov` container for example, maybe then it doesn't work.

![screen shot 2015-06-16 at 21 49 22](https://cloud.githubusercontent.com/assets/30694/8192949/d0d89798-1471-11e5-9eb5-c4b1221ec42d.png)
![screen shot 2015-06-16 at 21 49 13](https://cloud.githubusercontent.com/assets/30694/8192950/d0f34610-1471-11e5-989e-0e31d4c90ada.png)

**Comment by MichelvanderAa** [comment]
I can only encode via Quicktime WebM libraries and that always fits them into a .mov wrapper

The only other option I found online is that FFmpeg app, no idea how and where to download... the link mentioned above just gives me a folder with weird stuff..

**Comment by MichelvanderAa** [comment]
Robbert, for your FFmpeg codec settings please try:
720 x 1098 pixels @ 2000 kbps for SD
1280 x 1950 pixels @ 5000 kbps for HD

These gave me similar quality and file size when trying with that Quicktime WebM codec. Thanks!

**Comment by robbert** [comment]
I've added these to thebookofsand.net:
http://cdn.bookofsand.nl/hd.1080.webm
http://cdn.bookofsand.nl/sd.720.webm

**Comment by MichelvanderAa** [comment]
ok looking good! Only firefox uses these right?

**Comment by MichelvanderAa** [comment]
also, are they on our s3 ?

**Comment by robbert** [comment]
Yes, although technically: all browsers that don't support the first choice (`video/mp4`) and do support the second choice (`video/webm`).

The `cdn.bookofsand.nl` domains are aliases for Amazon S3.

**Comment by MichelvanderAa** [comment]
ok I'll ask some of the people reporting problems with Firefox to try again!

**Comment by robbert** [comment]
They might have to clear their browser caches before it works.

**Comment by MichelvanderAa** [comment]
Eerste reactie, werkt nog niet (heeft cache geleegd):
"Het blijft steeds hetzelfde... "Bezig met downloaden the book of sand."... balk schuift op naar 99% of 100%? en springt dan terug naar 75% tot 99% en dat blijft zich herhalen..."
